File: sdk/c#/PicarioXPO.RenderAPI/Fluent/IFluentXpoUrlFactory.cs
namespace PicarioXPO.RenderAPI.Fluent
{
    /// <summary>
    /// Represents the Fluent XPO URL generator
    /// </summary>
    public interface IFluentXpoUrlFactory
    {
        /// <summary>
        /// Creates a new instance of the currently implemented <c>IXpoUrlGenerator</c> and wraps it
        /// inside the <c>IFluentXpoUrlGenerator</c>
        /// </summary>
        IFluentXpoUrlGenerator CreateFluentUrlGenerator(IXpoUrlGenerator generator, FluentXpoUrlType urlType);
    }
}

File: sdk/js/PicarioXPO.RenderAPI.V2/UrlGenerators/IFluentXpoUrlFactory.ts
module UrlGeneratorModule {
    
    export interface IFluentXpoUrlFactory {
        
        /*
         * Creates a new instance of the currently implemented <c>IXpoUrlGenerator</c> and wraps it
         * inside the <c>IFluentXpoUrlGenerator</c>
         */
        createFluentUrlGenerator(generator: IXpoUrlGenerator, urlType: FluentXpoUrlType): IFluentXpoUrlGenerator;
    }

}